### ISSUE TITLE
When using mHC in my ViT model, the loss appears as nan

### ISSUE THREAD

**Issue by potato-creator** [lucidrains/hyper-connections#7]
Here is my code:
```
import torch
from torch import nn
branch = nn.Linear(512, 512)
from hyper_connections import mHC
init_hyper_conn, expand_stream, reduce_stream = mHC.get_init_and_expand_reduce_stream_functions(4)
hyper_conn = init_hyper_conn(dim = 512)
residual = expand_stream(residual)
branch_input, add_residual = hyper_conn(residual)
branch_output = branch(branch_input)
residual = add_residual(branch_output)
residual = reduce_stream(residual)
```
I don't know whether if I use it correctly. Thanks.

**Comment by lucidrains** [comment]
@potato-creator do you want to try [this](https://github.com/lucidrains/hyper-connections/blob/main/tests/test_hyper_connections.py#L256)
